lib/charms/opensearch/v0/helper_charm.py

```
@@ -228,3 +231,44 @@ def mask_sensitive_information(cmd: str) -> str:
     pattern = re.compile(r"(-tspass\s+|-kspass\s+|-storepass\s+|-new\s+|pass:)(\S+)")
 
     return re.sub(pattern, r"\1" + "xxx", cmd)
+
+
+def update_grafana_dashboards_titles(charm: "OpenSearchBaseCharm") -> None:
+    """Update the titles in the specified directory to include the charm revision."""
+    revision = get_charm_revision(charm.model.unit)
+    path = charm.charm_dir / "src/grafana_dashboards"
+
+    for dashboard_path in path.iterdir():
+        if dashboard_path.is_file() and dashboard_path.suffix == ".json":
+            try:
+                _update_dashboard_title(revision, dashboard_path)
+            except (json.JSONDecodeError, IOError) as e:
+                logger.error("Failed to process %s: %s", dashboard_path.name, str(e))
+        else:
+            logger.warning("%s is not a valid JSON file", dashboard_path.name)
```

I don't think we need to handle these

---

comment still relevant - the exception handling is not needed if we target `opensearch.json` - as a matter of fact we should let it crash instead of masking the issue

we can safely call `_update_dashboard_title(revision, dashboard_path)` directly